src/main/java/org/dice_group/grp/grammar/digram/DigramHelper.java

```diff
@@ -70,6 +89,77 @@
 		// TODO Auto-generated method stub
 		return null;
 	}
+	
+	/**
+	 * Finds the digram occurrences revolving around a specific statement in a given model
+	 * @param graph
+	 * @param newStmt
+	 * @return
+	 */
+	public static Set<DigramOccurence> findStmtBasedDigrams(Model graph, Statement newStmt) {
```

Comment:
I if understand the algorithm correct, we need to search for new digrams as follows: 
if mfd is replaced by A with A as the non terminal: 
We need to search for new Digrams and their Occurences only as follows:

`?n1 A  ?o2 . ?n2 ?p ?n3`
And 
`?n1 ?p  ?o2 . ?n2 A ?n3`

As we have only non overlapping digrams this is sufficient. 
If i understand the function correct it searches for 
n1 A n2 . n2 ?p ?n3 
and so on. 
So if we replaced X digrams it will do the algorithm for all these X new Statements. Am i correct? 
If yes: sounds good. 

However we can profile later on against an approach where we directly search
with 
`?n1 A  ?o2 . ?n2 ?p ?n3`
And 
`?n1 ?p  ?o2 . ?n2 A ?n3`

Not sure which is faster.

Reply:
yes, it searches for the digram occurrences surrounding only the new statements with the same 3 cases as before :)

Reply:
`?o2` = `?n2`, right? Otherwise, it does not really make sense to me :thinking:

Reply:
yes, 
`?o2 = ?n2`
sorry about that.